lib/ansible/module_utils/network/vyos/config/interfaces/interfaces.py

```diff
@@ -0,0 +1,432 @@
+        commands = list()
+        warnings = list()
+
+        existing_interfaces_facts = self.get_interfaces_facts()
+        commands.extend(self.set_config(existing_interfaces_facts))
+        if commands:
+            if self._module.check_mode:
+                resp = self._connection.edit_config(commands, commit=False)
+            else:
+                resp = self._connection.edit_config(commands)
+            result['changed'] = True
+
+        result['commands'] = commands
+
+        if self._module._diff:
+            result['diff'] = resp['diff'] if result['changed'] else None
+
+        changed_interfaces_facts = self.get_interfaces_facts()
+
+        result['before'] = existing_interfaces_facts
+        if result['changed']:
+            result['after'] = changed_interfaces_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_interfaces_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_interfaces_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands.extend(Interfaces._state_overridden(want=want, have=have))
+        else:
+            for item in want:
+                name = item['name']
+                obj_in_have = search_obj_in_list(name, have)
+                if state == 'deleted':
+                    commands.extend(
+                        Interfaces._state_deleted(
+                            want_intf=item, have_intf=obj_in_have
+                        )
+                    )
+
+                elif state == 'merged':
+                    commands.extend(
+                        Interfaces._state_merged(
+                            want_intf=item, have_intf=obj_in_have
+                        )
+                    )
+
+                elif state == 'replaced':
+                    commands.extend(
+                        Interfaces._state_replaced(
+                            want_intf=item, have_intf=obj_in_have
+                        )
+                    )
+
+        return commands
+
+    @staticmethod
+    def _state_replaced(**kwargs):
```

Comment:
What is the reason to use kwargs here? Since the argument list is defined I don't see a reason to use kwargs. Applicable at other places as well in this file